File: TruSec.Backend/TruSec.DAL/Entities/TruckDataLog.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace TruSec.DAL.Entities
{
    public class TruckDataLog
    {
        public int Id { get; set; }
        public DateTime TimeStamp { get; set; }
        public string DriverExpression { get; set; }
        public double Latitude { get; set; }
        public double Longitude { get; set; }
        public int TruckId { get; set; }
        [ForeignKey(nameof(TruckId))]
        public Truck Truck { get; set; }
        public double SpeedKPH { get; set; }
    }
}

File: TruSec.WebClient/src/app/models/TruckDataLog.ts
import { Truck } from "./Truck";


export class TruckDataLog {
    id?: number;
    timeStamp?: Date;
    driverExpression?: string;
    latitude?: number;
    longitude?: number;
    truckId?: number;
    truck?: Truck;
    speedKPH?: number;
}
